src/sync/parallel-page-fetcher.ts

```diff
@@ -0,0 +1,45 @@
+import { TaskPayload } from "~/src/sync/sync.types";
+
+const mergeJiraPayload = (jiraPayload1?: any, jiraPayload2?: any) => {
+	if (!jiraPayload1 && !jiraPayload2) {
+		return undefined;
+	}
+
+	// TODO: add more
+	const pullRequests = [...(jiraPayload1?.pullRequests || []), ...(jiraPayload2?.pullRequests || [])];
+	const builds = [...(jiraPayload1?.builds || []), ...(jiraPayload2?.builds || [])];
+
+	if (pullRequests.length === 0 && builds.length === 0) {
+		return undefined;
+	}
+
+	return {
+		...(jiraPayload1 || jiraPayload2),
+		... (pullRequests.length > 0 ? { pullRequests } : {}),
```

Comment:
Nit: remove the space before ... Doesn't really matter but just figured we don't have a space for the first spread operator but do for the 2nd 🤷